src/chains/ethereum/ethereum/tests/provider.test.ts

```diff
@@ -141,6 +141,84 @@ describe("provider", () => {
       await provider.disconnect();
     });
 
+    describe("uses timestamp adjustment in subsequent blocks after calling `evm_mine` with a `timestamp` argument", () => {
+      const timestampIncrement = 5; // seconds
+      const timeArgumentSeconds = 100;
+
+      async function mineBlocksForTimestamps(
+        provider: EthereumProvider
+      ): Promise<string[]> {
+        // mine a block with a specified timestamp
+        await provider.request({
+          method: "evm_mine",
+          params: [timeArgumentSeconds]
+        });
+        const specifiedBlock = await provider.request({
+          method: "eth_getBlockByNumber",
+          params: ["latest", false]
+        });
+        // mine a block without a specified timestamp
+        await provider.request({
+          method: "evm_mine",
+          params: []
+        });
+        const unspecifiedBlock = await provider.request({
+          method: "eth_getBlockByNumber",
+          params: ["latest", false]
+        });
+
+        return [specifiedBlock.timestamp, unspecifiedBlock.timestamp];
+      }
+
+      it("should work with `timestampIncrement` of `clock`", async () => {
+        const provider = await getProvider({
+          miner: { timestampIncrement: "clock" }
+        });
+
+        const [specifiedBlock, unspecifiedBlock] =
+          await mineBlocksForTimestamps(provider);
+
+        assert.strictEqual(
+          specifiedBlock,
+          `0x${timeArgumentSeconds.toString(16)}`,
+          "Unexpected timestamp for block mined with specified timestamp"
+        );
+
+        assert(
+          +unspecifiedBlock >= timeArgumentSeconds &&
+            +unspecifiedBlock <= timeArgumentSeconds + 1000,
```

Comment:
If CI is running slowly, could this break? I don't know what the solution is, and I don't feel it's worth holding up this PR for, but just wanted to throw it out there in case you have a solution on-hand.

Reply:
It _could_ do. Because we are specifying a timestamp in seconds, it would need to take an entire second to mine and fetch 2 blocks.

I feel like I'm building a lot of pressure on the time refactor - but it'll let us inject a `now` provider, which allow us to be much more resilient to slow build agents.

🤞